Subject: [PATCH] Holiday patch #1
Add infra extension to IdentityService
---
Index: src/services/trips/TripAhead.Trips.Infrastructure/DependencyInjection.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/services/trips/TripAhead.Trips.Infrastructure/DependencyInjection.cs b/src/services/trips/TripAhead.Trips.Infrastructure/DependencyInjection.cs
--- a/src/services/trips/TripAhead.Trips.Infrastructure/DependencyInjection.cs	(revision 1b7a6da451b245121f5be4a00455c19942efdfab)
+++ b/src/services/trips/TripAhead.Trips.Infrastructure/DependencyInjection.cs	(date 1710574733077)
@@ -4,7 +4,6 @@
 using TripAhead.Trips.Application.Features.Trips.Queries;
 using TripAhead.Trips.Domain.Repositories;
 using TripAhead.Trips.Infrastructure.DataAccess;
-using TripAhead.Trips.Infrastructure.Persistance;
 using TripAhead.Trips.Infrastructure.Persistence;
 
 namespace TripAhead.Trips.Infrastructure;
Index: src/services/identity/TripAhead.Identity.API/DependencyInjection.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/services/identity/TripAhead.Identity.API/DependencyInjection.cs b/src/services/identity/TripAhead.Identity.API/DependencyInjection.cs
new file mode 100644
--- /dev/null	(date 1710574733072)
+++ b/src/services/identity/TripAhead.Identity.API/DependencyInjection.cs	(date 1710574733072)
@@ -0,0 +1,33 @@
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace TripAhead.Identity.API;
+
+public static class DependencyInjection
+{
+    public static IServiceCollection AddInfrastructure(this IServiceCollection services, IConfiguration configuration)
+    {
+        services
+            .AddServices()
+            .AddPersistance(configuration);
+
+        return services;
+    }
+
+    private static IServiceCollection AddServices(this IServiceCollection services)
+    {
+        services.AddMediatR(cfg =>
+            cfg.RegisterServicesFromAssembly(typeof(Program).Assembly));
+
+        return services;
+    }
+
+    private static IServiceCollection AddPersistance(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddDbContext<IdentityDbContext>(options =>
+            options.UseNpgsql(configuration.GetConnectionString("IdentityContext")));
+        
+        return services;
+    }
+}
\ No newline at end of file
Index: src/services/identity/TripAhead.Identity.API/Program.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/services/identity/TripAhead.Identity.API/Program.cs b/src/services/identity/TripAhead.Identity.API/Program.cs
--- a/src/services/identity/TripAhead.Identity.API/Program.cs	(revision 1b7a6da451b245121f5be4a00455c19942efdfab)
+++ b/src/services/identity/TripAhead.Identity.API/Program.cs	(date 1710574573571)
@@ -1,4 +1,5 @@
 using Microsoft.AspNetCore.Identity;
+using TripAhead.Identity.API;
 using TripAhead.Identity.Api.DataAccess;
 using TripAhead.Identity.Api.Models;
 
@@ -13,6 +14,8 @@
         .AddEntityFrameworkStores<ApplicationDbContext>()
         .AddDefaultTokenProviders();
 
+builder.Services.AddInfrastructure(builder.Configuration);
+
 // Add services to the container.
 builder.Services.AddProblemDetails();
 
Index: src/services/identity/TripAhead.Identity.API/TripAhead.Identity.API.csproj
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/services/identity/TripAhead.Identity.API/TripAhead.Identity.API.csproj b/src/services/identity/TripAhead.Identity.API/TripAhead.Identity.API.csproj
--- a/src/services/identity/TripAhead.Identity.API/TripAhead.Identity.API.csproj	(revision 1b7a6da451b245121f5be4a00455c19942efdfab)
+++ b/src/services/identity/TripAhead.Identity.API/TripAhead.Identity.API.csproj	(date 1710574425741)
@@ -19,6 +19,7 @@
     </ItemGroup>
 
     <ItemGroup>
+      <ProjectReference Include="..\..\..\libs\TripAhead.Common\TripAhead.Common.csproj" />
       <ProjectReference Include="..\..\..\TripAhead.ServiceDefaults\TripAhead.ServiceDefaults.csproj" />
     </ItemGroup>
 
Index: src/services/trips/TripAhead.Trips.Infrastructure/Persistence/OptionalItemRepository.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/services/trips/TripAhead.Trips.Infrastructure/Persistence/OptionalItemRepository.cs b/src/services/trips/TripAhead.Trips.Infrastructure/Persistence/OptionalItemRepository.cs
--- a/src/services/trips/TripAhead.Trips.Infrastructure/Persistence/OptionalItemRepository.cs	(revision 1b7a6da451b245121f5be4a00455c19942efdfab)
+++ b/src/services/trips/TripAhead.Trips.Infrastructure/Persistence/OptionalItemRepository.cs	(date 1710574733080)
@@ -3,7 +3,7 @@
 using TripAhead.Trips.Domain.Repositories;
 using TripAhead.Trips.Infrastructure.DataAccess;
 
-namespace TripAhead.Trips.Infrastructure.Persistance;
+namespace TripAhead.Trips.Infrastructure.Persistence;
 
 public class OptionalItemRepository(TripsDbContext dbContext) : IOptionalItemRepository
 {
